### DISCUSSION TITLE
How to use THERMOSTAT_CONTROLLER_OUTPUTS?

### DISCUSSION THREAD

**Discussion by gck303** [arendst/Tasmota#11960] [Q&A] [answered]
How do I use multiple thermostats within a single Tasmota? I have compiled a custom 9.4.0 tasmota binary and do not understand how it works. 

#define THERMOSTAT_CONTROLLER_OUTPUTS         4

I would have thought that the commands would be as below, but it does not seem to work:
TEMPTARGETSET1
TEMPTARGETSET2
TEMPTARGETSET3
TEMPTARGETSET4

I have looked in the instructions and checked the other forums, but cannot find anything. 

And then, once I have got the four thermostats working I need to reference the TARGET and MEASURED temperatures in a script. How do I do that?

Many thanks, George

**Comment by gck303** [comment, marked as answer]
Okay. My initial idea works.

You can see the index on the response to confirm it has changed. 

`20:54:55.695 CMD: TEMPTARGETSET2 20 
20:54:55.699 SRC: WebConsole from 192.168.0.38
20:54:55.702 CMD: Group 0, Index 2, Command "TEMPTARGETSET", Data "20 "
20:54:55.709 MQT: stat/d1_1/RESULT = {"TempTargetSet":20.0}`

**Reply by MrNomNom** [reply]
Yes it works like this, but annoying thing here is that result response doesnt indicate which controller requested data.

eg. When I receive “ 20:54:55.709 MQT: stat/d1_1/RESULT = {"TempTargetSet":20.0}” i have no idea which controller this belongs to.

**Reply by ascillato** [reply]
Yes, you are right.
You can change the code in the TempTargetSet command response to add the index like stat/d1_1/RESULT = {"TempTargetSet2":20.0}

**Reply by MrNomNom** [reply]
I did exactly this in my build :) Still not sure if I should make a merge request, as I am not very familiar with codebase to be sure I am not breaking stuff for people...

**Reply by ascillato** [reply]
Please, do. We can review it and help if there is something to fix. Thanks.
